### DISCUSSION TITLE
How to export a .mo model to .fmu inOpenModelica (nightly builds for windows  1.23.0-dev)

### DISCUSSION THREAD

**Discussion by ghost** [OpenModelica/OpenModelica#11750] [Q&A]
Dear developers:

I downloaded and installed without any issues [this ](https://openmodelica.org/download/download-windows/)OpenModelica in my computer windows x64.
Also I have downloaded a typical .mo model (BouncingBall.mo) from a public repository and it works perfectly fine in OpenModelica
However, when I try exporting it through the GUI, it gives me the following error and I do ot know why occurs and what is going on:
![image](https://github.com/OpenModelica/OpenModelica/assets/154359407/29190aaa-5223-437e-8a97-13faf24ef76d)

Can anyone help me with this issue?

**Comment by casella** [comment]
What do you mean exactly with "exporting it through the GUI"? Which menu items and setup options did you use?

**Reply by ghost** [reply]
Thank you
What I mean is that I open OpenModelica with double-click, I open my model, then right click to my model and then select export>FMU
![image](https://github.com/OpenModelica/OpenModelica/assets/154359407/6b09596d-0ef6-405d-9d8f-3e261da40695)

**Comment by casella** [comment]
Did you try the latest OpenModelica nightly build?

**Reply by ghost** [reply]
I am using the 1.23.0-dev Nightly Build downlaoded from [here](https://openmodelica.org/download/download-windows/).
However I also tried with a previous version and had the exact same "Translation error" isue when trying to export as FMU

**Comment by arun3688** [comment]
@casella I Tested the latest omc version and i also get the same error, This issue is related with this issue https://github.com/OpenModelica/OpenModelica/issues/11730  and i guess it is windows related issue.

**Reply by casella** [reply]
OK. Do you think you can manage to fix #11554 as suggested there? It's just a matter of changing the file names of the generated code when building FMUs

**Reply by ghost** [reply]
@arun3688 Actually I have another PC (with ubuntu-linux) in which I have kinda same "Translation Error" issue when trying to do the same:
![erroropen](https://github.com/OpenModelica/OpenModelica/assets/154359407/77a9ab7b-d8b4-481c-a841-48fa5afc98ae)

**Comment by arun3688** [comment]
@casella the issue is not related to `longPath` or `filenames` but something else in the build process of FMU with `cmake` i guess see the error below , An example mos script to try 

```
loadModel(Modelica);
getErrorString();

buildModelFMU(Modelica.Electrical.Analog.Examples.CauerLowPassAnalog, fileNamePrefix="test");
getErrorString();
```

```
"Error: Error building simulator. Build log: -- The C compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: C:/OMDev/tools/msys/mingw64/bin/gcc.exe
-- Check for working C compiler: C:/OMDev/tools/msys/mingw64/bin/gcc.exe - works
-- Detecting C compile features
-- Detecting C compile features - done
-- FMI2 include directory: C:/OPENMODELICAGIT/OpenModelica/install/include/omc/c/fmi
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Error at CMakeLists.txt:95 (if):
  if given arguments:

    \"EQUAL\" \"8\"

  Unknown arguments specified


-- Configuring incomplete, errors occurred!
See also \"C:/OPENMODELICAGIT/test/test.fmutmp/sources/build_cmake_dynamic/CMakeFiles/CMakeOutput.log\".
See also \"C:/OPENMODELICAGIT/test/test.fmutmp/sources/build_cmake_dynamic/CMakeFiles/CMakeError.log\".
"

```

**Comment by adrpo** [comment]
This is the Windows Defender thinking that CMake generated executables are viruses (false positives).

![image](https://github.com/OpenModelica/OpenModelica/assets/4192979/75c47912-8046-4378-bbde-c440ecf16c6a)

![image](https://github.com/OpenModelica/OpenModelica/assets/4192979/e80446ec-207d-476a-a57b-84a122afbb13)

Unfortunately I have no idea how to fix this until Microsoft gets its act together and gets rid of these false positives.

It seems some other people have this issue with no resolution:
https://discourse.cmake.org/t/windows-defender-blocked-a-exe-when-configuring/9647
https://youtrack.jetbrains.com/issue/CPP-11642/WINDOWS-DEFENDER-IDENTIFIES-CMAKE-FILES-AS-VIRUS-BREAKS-CLION-COMPILATION

I sent some of these files to Microsoft so they can check is not a virus, but they haven't fixed this yet.

**Reply by casella** [reply]
@vissari17 you may try adding OpenModelica's working directory to the exclusion list of the antivirus.

**Comment by adrpo** [comment]
You can also use the OpenModelica release based on the newer msys2 UCRT:
https://build.openmodelica.org/omc/builds/windows/nightly-builds/PRbuilds/64bit/
which doesn't seem to trigger the antivirus.
